warehouse/query-core/src/main/java/datawave/query/postprocessing/tf/TermFrequencyHitFunction.java

```diff
@@ -0,0 +1,689 @@
+                    negativeSubQueries.add(subQuery);
+                    for (String fieldValue : subQuery.fieldValues) {
+                        fvToFunctionIds.put(fieldValue, subQuery.id);
+                        negativeSearchSpace.add(fieldValue);
+                    }
+                } else {
+                    positiveSubQueries.add(subQuery);
+                    for (String fieldValue : subQuery.fieldValues) {
+                        fvToFunctionIds.put(fieldValue, subQuery.id);
+                        positiveSearchSpace.add(fieldValue);
+                    }
+                }
+            }
+        }
+    }
+    
+    /**
+     * Get this function's search space as defined by fields and values
+     *
+     * @param function
+     *            a {@link Function} that is either adjacent, phrase, or within
+     * @return a list of fields and values associated with the provided function
+     */
+    private List<SubQuery> parseFunction(Function function) {
+        // functions may take different forms..
+        // within = {field, number, termOffsetMap, terms...}
+        // within = {number, termOffsetMap, terms...}
+        // adjacent = {field, termOffsetMap, terms...}
+        // adjacent = {termOffsetMap, terms...}
+        // phrase = {field, termOffsetMap, terms...}
+        // phrase = {termOffsetMap, terms...}
+        
+        if (!TermOffsetPopulator.phraseFunctions.contains(function.name())) {
+            log.error(function.name() + " is not a supported content function.");
+            return Collections.EMPTY_LIST;
+        }
+        
+        List<JexlNode> args = function.args();
+        int index = function.name().equals("within") ? 3 : 2;
+        
+        // If the first arg is a number or termOffsetMap then no field was provided as part of the function.
+        // The fields will be build via value lookups from the tfFVs.
+        boolean specialCase = false;
+        JexlNode first = args.get(0);
+        if (isFirstNodeSpecial(first)) {
+            specialCase = true;
+            index--;
+        }
+        
+        // Parse the values first. Might have to lookup fields by value.
+        TreeSet<String> values = getValuesFromArgs(args, index);
+        
+        TreeSet<String> fields;
+        if (specialCase) {
+            // field(s) were not present in the function, lookup fields by value.
+            fields = lookupFieldsByValues(values);
+        } else {
+            // function's first arg is the field, or fields in the form (FIELD_A || FIELD_B)
+            fields = parseField(args.get(0));
+        }
+        
+        // nodes under a delayed marker are not part of the field index lookup and do not have values in the document. Values from a content function delayed
+        // for any reason must be fetched by this function.
+        boolean isNegated = isDelayed(function);
+        if (isNegated) {
+            this.hasNegatedFunctions = true;
```

Comment:
Should this be named as "hasDelayedFunctions"? this.hasDelayedFunctions = isDelayed(function) ??

Reply:
Seems reasonable, updated variables and comment for clarity